src/workspacefactory/wfactory_generator.js

```diff
@@ -163,26 +228,38 @@ WorkspaceFactoryGenerator.prototype.generateInjectString = function() {
 
   var attributes = addAttributes(this.model.options, '\t');
   if (!this.model.options['readOnly']) {
-    attributes = '\ttoolbox : toolbox, \n' +
-      attributes;
+    attributes = '\ttoolbox : BLOCKLY_TOOLBOX_XML[/* TODO: Insert name of ' +
+      'imported toolbox to display here */], \n' + attributes;
   }
-  var finalStr = '/* TODO: Change toolbox XML ID if necessary. Can export ' +
-      'toolbox XML from Workspace Factory. */\n' +
-      'var toolbox = document.getElementById("toolbox");\n\n';
+
+  // Initializing toolbox
+  var xmlString = `/* DO NOT EDIT THE CODE BELOW */
+if(!BLOCKLY_TOOLBOX_XML) {
+  var BLOCKLY_TOOLBOX_XML = {};
+}
+/* DO NOT EDIT THE CODE ABOVE */`;
+
+  var finalStr = xmlString + '\n\n';
   finalStr += 'var options = { \n' + attributes + '};';
   finalStr += '\n\n/* Inject your workspace */ \nvar workspace = Blockly.' +
       'inject(/* TODO: Add ID of div to inject Blockly into */, options);';
-  finalStr += '\n\n/* Load Workspace Blocks from XML to workspace. ' +
-      'Remove all code below if no blocks to load */\n\n' +
-      '/* TODO: Change workspace blocks XML ID if necessary. Can export' +
-      ' workspace blocks XML from Workspace Factory. */\n' +
-      'var workspaceBlocks = document.getElementById("workspaceBlocks"); \n\n' +
-      '/* Load blocks to workspace. */\n' +
-      'Blockly.Xml.domToWorkspace(workspace, workspaceBlocks);';
   return finalStr;
 };
 
 /**
+ * Converts one-line XML string to multi-line string concatenation expression.
+ * Used for making XML string more readable.
+ *
+ * @returns {string} XML string as string concatenation without unnecessary
+ * whitespace between tags.
+ * @private
+ */
+WorkspaceFactoryGenerator.prototype.splitXmlWithNewline_ = function(xmlString) {
+  var totalString = xmlString.replace(/>( |\n)*</g, '>\' +\n    \'<');
```

Comment:
Escape the XML inside the JS strings.  I can currently break this by including a single quote in any of my names.